### PR TITLE
Catch gdb.error

### PR DESCRIPTION
**Pull request by Reinis**
Seen when debugging loadkeys from kdb package.

### CODE REVIEW

**Comment by cyrus-and** [comment]
Thanks, do you have any simple way for me to trigger the error?

File: .gdbinit [hunk 1]
```diff
@@ -191,6 +191,8 @@ def to_string(value):
         value_string = str(value)
     except UnicodeEncodeError:
         value_string = unicode(value).encode('utf8')
+    except gdb.error as err:
+        value_string = '<gdb.error: {}>'.format(str(err))
```
Comment (cyrus-and) [inline .gdbinit]: Use the following to be more coherent with the existing code:

```python
    except gdb.error as e:
        value_string = ansi(e, R.style_error)
```

**Comment by Reinis** [comment]
> Thanks, do you have any simple way for me to trigger the error?

In VT (e.g. Ctrl+Alt+F3):
```
$ gdb loadkeys
GNU gdb (Gentoo 8.3.50.20190312 p1) 8.3.50.20190312-git
...
>>> dashboard -output /dev/pts/7
>>> dashboard stack -style locals True
>>> break yyparse
>>> set args -pv us
>>> run
```
```
─── Stack ────────────────────────────────────────
Traceback (most recent call last):
  File "<string>", line 403, in render
  File "<string>", line 1175, in lines
  File "<string>", line 1210, in fetch_frame_info
  File "<string>", line 210, in format_value
  File "<string>", line 190, in to_string
gdb.error: value requires 104000 bytes, which is more than max-value-size
```

**Comment by cyrus-and** [comment]
Awesome, thanks!

**Pull request merged**